Content.Server/Botany/Systems/BotanySwabSystem.cs

```diff
@@ -71,12 +88,55 @@ private void OnDoAfter(EntityUid uid, BotanySwabComponent swab, DoAfterEvent arg
             var old = plant.Seed;
             if (old == null)
                 return;
+
             plant.Seed = _mutationSystem.Cross(swab.SeedData, old); // Cross-pollenate
-            swab.SeedData = old; // Transfer old plant pollen to swab
+
+            if (swab.Contaminate)
+                swab.SeedData = old;// Transfer old plant pollen to swab if contamination is allowed
+
             _popupSystem.PopupEntity(Loc.GetString("botany-swab-to"), args.Args.Target.Value, args.Args.User);
         }
+        args.Handled = true;
+    }
+
+    private void OnClean(EntityUid uid, BotanySwabComponent swab, ref UseInHandEvent args)
+    {
+        if (args.Handled)
+            return;
+
+        if (!swab.Cleanable)
+            return;
 
+        swab.SeedData = null;
+        _popupSystem.PopupEntity(Loc.GetString("botany-swab-clean"), uid, args.User);
+        _audioSystem.PlayPvs(swab.CleanSound, uid);
         args.Handled = true;
     }
+
+    private void OnInsertAttempt(EntityUid uid, BotanySwabComponent swab, ref ContainerGettingInsertedAttemptEvent args)
+    {
+        //does the container have the botanySwab component
+        if (!HasComp<BotanySwabComponent>(args.Container.Owner))
```

Comment:
Wouldn't this make it impossible to put swabs into a container? I think you need to change your logic here.
```suggestion
        if (swab.SeedData == null && HasComp<BotanySwabComponent>(args.Container.Owner))
```

Reply:
The function cancels if its criteria of the owner (the swab applicator) not having the BotanySwabComponent and the swab not being null aren't met. It returning means that the insert attempt "passes" and continues as it should.

In game this is seen with trying to put a clean swab into a swab applicator causing it to pop back out due to clean swabs not being accepted. Used ones will be accepted just fine.